vm/src/obj/objellipsis.rs

```diff
@@ -18,3 +19,8 @@ fn ellipsis_repr(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
     arg_check!(vm, args, required = [(_cls, None)]);
     Ok(vm.new_str("Ellipsis".to_string()))
 }
+
+fn ellipsis_reduce(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(vm, args, required = [(_cls, None)]);
+    Ok(vm.new_str("Ellipsis".to_string()))
+}
```

Comment:
I know this is a really simple function, but it's good to know how to write RustPython functions that can take advantage of the `Fn()` type shenanigans we have set up to make the signatures more readable.

First off, it looks like `Ellipsis.__reduce__` does require an `Ellipsis` as its self argument:
```
>>> type(Ellipsis).__reduce__(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor '__reduce__' requires a 'ellipsis' object but received a 'int'
```

So we want to write a function that takes one parameter and returns a string. We can do that by just writing a rust function that expresses that signature, because we know all the types:

```rust
fn ellipsis_reduce(
    _self: PyEllipsisRef, // from crate::pyobject
    _vm: &VirtualMachine, // necessary to mark that this is a python function
) -> String {
    "Ellipsis".to_string()
}

All the argument checking is done by the `FromArgs` trait in `function.rs`, and the result conversion is done by the `IntoPyObject` trait from `pyobject.rs`.